scalafix-rules/src/main/scala/scalafix/internal/rule/ExplicitResultTypes.scala

```diff
@@ -18,11 +18,14 @@ final class ExplicitResultTypes(
 ) extends SemanticRule("ExplicitResultTypes") {
 
   def this() = this(ExplicitResultTypesConfig.default, LazyValue.now(None))
+  val supportedScalaVersions = RulesBuildInfo.supportedScalaVersions
+  val runtimeScalaVersion = RulesBuildInfo.scalaVersion
```

Comment:
As far as I remember, I suggested the name `runtimeScalaVersion` because you were using `scala.util.Properties.versionNumberString` in an earlier commit of this PR. Now that it's using `RulesBuildInfo`, I think the right naming would be
```suggestion
  val compilerScalaVersion = RulesBuildInfo.scalaVersion
```

In some rare cases, I guess the runtime can differ from that. But from my understanding anyway, this is actually closer to what we want: compare the presentation compiler version brought by scalameta (using the exact same scalaVersion as `rules`) against the version of scalac used for the input.

Am I getting this right @olafurpg ?

Reply:
Anyway, this is just naming so it can be tweaked later (maybe in the same PR as the previous set of cosmetic suggestions I had), I just wanted to make sure we were doing/testing the right thing here.

Reply:
> against the version of scalac used for the input

JFYI, there's also `scala.tools.nsc.Properties.versionNumberString` which is the compiler's version, but it's highly unlikely to vary compared to the standard library version.

Reply:
> it's highly unlikely to vary compared to the standard library version.

I guess that can only happen if the scala version of scalafix-core & scalameta drift, which is rarely the case (https://github.com/scalacenter/scalafix/pull/1179/files bumps both for example)
```
$ cs resolve -t ch.epfl.scala::scalafix-rules:0.9.17
  Result:
└─ ch.epfl.scala:scalafix-rules_2.13:0.9.17
   ├─ ch.epfl.scala:scalafix-core_2.13:0.9.17
   │  ├─ com.geirsson:metaconfig-typesafe-config_2.13:0.9.10
   │  │  ├─ com.geirsson:metaconfig-core_2.13:0.9.10
   │  │  │  ├─ com.lihaoyi:pprint_2.13:0.5.9
   │  │  │  │  ├─ com.lihaoyi:fansi_2.13:0.2.9
   │  │  │  │  │  └─ com.lihaoyi:sourcecode_2.13:0.2.1
   │  │  │  │  └─ com.lihaoyi:sourcecode_2.13:0.2.1
   │  │  │  ├─ org.scala-lang:scala-library:2.13.1 -> 2.13.2
   │  │  │  ├─ org.scala-lang.modules:scala-collection-compat_2.13:2.1.2 -> 2.1.6
   │  │  │  │  └─ org.scala-lang:scala-library:2.13.1 -> 2.13.2
   │  │  │  └─ org.typelevel:paiges-core_2.13:0.3.0
   │  │  │     └─ org.scala-lang:scala-library:2.13.0 -> 2.13.2
   │  │  ├─ com.typesafe:config:1.2.1
   │  │  └─ org.scala-lang:scala-library:2.13.1 -> 2.13.2
   │  ├─ com.googlecode.java-diff-utils:diffutils:1.3.0
   │  ├─ org.scala-lang:scala-library:2.13.2
   │  ├─ org.scala-lang.modules:scala-collection-compat_2.13:2.1.6
   │  │  └─ org.scala-lang:scala-library:2.13.1 -> 2.13.2
   │  └─ org.scalameta:scalameta_2.13:4.3.14
   │     ├─ org.scala-lang:scala-library:2.13.2
   │     ├─ org.scala-lang:scalap:2.13.2
   │     │  └─ org.scala-lang:scala-compiler:2.13.2
   │     │     ├─ net.java.dev.jna:jna:5.3.1
   │     │     ├─ org.jline:jline:3.14.1
   │     │     ├─ org.scala-lang:scala-library:2.13.2
   │     │     └─ org.scala-lang:scala-reflect:2.13.2
   │     │        └─ org.scala-lang:scala-library:2.13.2
   │     └─ org.scalameta:parsers_2.13:4.3.14
   │        ├─ org.scala-lang:scala-library:2.13.2
   │        └─ org.scalameta:trees_2.13:4.3.14
   │           ├─ com.thesamet.scalapb:scalapb-runtime_2.13:0.10.3
   │           │  ├─ com.google.protobuf:protobuf-java:3.11.4
   │           │  ├─ com.lihaoyi:fastparse_2.13:2.3.0
   │           │  │  ├─ com.lihaoyi:geny_2.13:0.6.0
   │           │  │  └─ com.lihaoyi:sourcecode_2.13:0.2.1
   │           │  ├─ com.thesamet.scalapb:lenses_2.13:0.10.3
   │           │  │  ├─ org.scala-lang:scala-library:2.13.1 -> 2.13.2
   │           │  │  └─ org.scala-lang.modules:scala-collection-compat_2.13:2.1.6
   │           │  │     └─ org.scala-lang:scala-library:2.13.1 -> 2.13.2
   │           │  ├─ org.scala-lang:scala-library:2.13.1 -> 2.13.2
   │           │  └─ org.scala-lang.modules:scala-collection-compat_2.13:2.1.6
   │           │     └─ org.scala-lang:scala-library:2.13.1 -> 2.13.2
   │           ├─ org.scala-lang:scala-library:2.13.2
   │           ├─ org.scalameta:common_2.13:4.3.14
   │           │  ├─ com.lihaoyi:sourcecode_2.13:0.2.1
   │           │  └─ org.scala-lang:scala-library:2.13.2
   │           └─ org.scalameta:fastparse_2.13:1.0.1
   │              ├─ com.lihaoyi:sourcecode_2.13:0.1.7 -> 0.2.1 (possible incompatibility)
   │              ├─ org.scala-lang:scala-library:2.13.0 -> 2.13.2
   │              └─ org.scalameta:fastparse-utils_2.13:1.0.1
   │                 ├─ com.lihaoyi:sourcecode_2.13:0.1.7 -> 0.2.1 (possible incompatibility)
   │                 └─ org.scala-lang:scala-library:2.13.0 -> 2.13.2
   ├─ org.scala-lang:scala-library:2.13.2
```

Reply:
It's impossible in sbt without disabling things as it has this `OverrideScalaMediator` (https://github.com/sbt/sbt/pull/2634) which keeps scala-compiler/scala-library/scala-reflect in sync.  And then there's are also, in the compiler, various sanity checks on the classes in the library, so it crashes early if you mix major versions for example.